### DISCUSSION TITLE
check updates on boot

### DISCUSSION THREAD

**Discussion by LuckyTurtleDev** [containrrr/watchtower#1067] [Q&A] [answered]
It is possible to update only images at the start of watchtower?

I am using a docker image for my 3d-printer web-interface. If watchtower does update this image in the middle of a print, the print is lost. But I sill would like to have automatic updates. Because of this it would be nice, to update the image only at the start of the pc.

**Comment by tammert** [comment, marked as answer]
I don't think you can do this natively with Docker, but I did come up with something that could work? If you start `watchtower` with the `--run-once` flag, it'll just run once and exit with statuscode 0. You'd need to use an OS-based capability to make sure to start Watchtower again after a boot (with `docker start watchtower`). Unfortunately, a combination of `--run-once` with the Docker `--restart=something` flag will just lead to constant restarts.

Depending on your setup, an entry into `.bashrc` or `.zshrc` could do the trick, or you could create a systemd service as described [here](https://askubuntu.com/questions/919054/how-do-i-run-a-single-command-at-startup-using-systemd) (if you have a systemd based OS).

**Comment by LuckyTurtleDev** [comment]
Thanks

I have create a watchtower cronjob by adding the following line to the `/etc/crontab`:
```
@reboot root sleep 15 && docker run --rm --name watchtower -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower --run-once --include-restarting --cleanup > /var/log/cron-watchtower.log 2>&1
```

please consider, that `@reboot` is not part of the cron standard and is not supported by all systems.
